prerendering.bs

```diff
@@ -259,23 +304,48 @@ Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which
+  1. [=list/For each=] |record| of |referrerDoc|'s [=Document/prerender records=]:
+      1. If |record|'s [=prerender record/starting URL=] is |startingURL|, then return.
 
   1. Let |prerenderingTraversable| be the result of [=creating a new top-level traversable=].
 
   1. Set |prerenderingTraversable|'s [=navigable/loading mode=] to "`prerender`".
 
-  1. Set |referrerDoc|'s [=Document/prerendering traversables map=][|startingURL|] to |prerenderingTraversable|.
+  1. Let |prerenderRecord| be a [=prerender record=] with [=prerender record/starting URL=] |startingURL|, [=prerender record/No-Vary-Search hint=] |nvsHint|, [=prerender record/start time=] the [=current high resolution time=] for the [=relevant global object=] of |referrerDoc|, and [=prerender record/prerendering traversable=] |prerenderingTraversable|.
 
-  1. Set |prerenderingTraversable|'s [=prerendering traversable/remove from referrer=] to be an algorithm which [=map/removes=] |referrerDoc|[|startingURL|].
+  1. [=list/Append=] |prerenderRecord| to |referrerDoc|'s [=Document/prerender records=].
+
+  1. Set |prerenderingTraversable|'s [=prerendering traversable/remove from referrer=] to be an algorithm which [=list/removes=] |prerenderRecord| from |referrerDoc|'s [=Document/prerender records=].
 
      <p class="note">As with all [=top-level traversables=], the [=prerendering traversable=] can be [=destroy a top-level traversable|destroyed=] for any reason, for example if it becomes unresponsive, performs a restricted operation, or if the user agent believes prerendering takes too many resources.
 
   1. [=Navigate=] |prerenderingTraversable| to |startingURL| using |referrerDoc|, with <i>[=navigate/referrerPolicy=]</i> set to |referrerPolicy|.
 </div>
 
 <div algorithm>
-  To <dfn for="prerendering traversable">activate</dfn> a [=prerendering traversable=] |successorTraversable| in place of a [=top-level traversable=] |predecessorTraversable| given a [=history handling behavior=] |historyHandling| and an optional [=navigation ID=] |navigationId|:
+
+  To <dfn for="prerendering traversable">update the successor for activation</dfn> given a [=prerendering traversable=] |successorTraversable|, a [=URL=] |startingURL|, and a [=URL=] |activationURL|:
+
+  1. Let |successorDocument| be |successorTraversable|'s [=navigable/active document=].
+
+  1. If |startingURL| equals |successorDocument|'s [=Document/URL=] and |startingURL| does not equal |activationURL|:
+
+    1. [=Assert=]: |successorDocument| [=can have its URL rewritten=] to |activationURL|.
+
+    1. Let |navigation| be |successorDocument|'s [=relevant global object=]'s [=navigation API=].
+
+    1. Let |continue| be the result of [=firing a push-replace-reload navigate event=] at |navigation| given "[=NavigationType/replace=]", |activationURL|, and true.
+       <p class="issue">Should we do this or just do the update without the event? TODO: resolve before submit
```

Comment:
@domenic in particular, for thoughts on whether a navigate event is appropriate here.

Reply:
I feel moderately strongly that a navigate event is a good idea here. (Haven't looked at the rest of the PR yet.)

Reply:
I agree canceling it would be a bit strange. However I think there's no reason to disallow that, similar to how we don't prevent a page from using `history.replaceState()` on activation to change its URL "back".

Reply:
The latest patch fires the event and removes this TODO.